### PR TITLE
Show delivery failure reasons if provided by Mailgun

### PR DESCRIPTION
**Pull request by fleetwoods**

### CODE REVIEW

File: app/models/Comm.scala [hunk 1]
```diff
@@ -166,8 +166,16 @@ object Comm {
...
 
-  private def findFailed(dbModel: DbModel) = dbModel.events.collectFirst {
-    case FailedEvent(event) => event
+  private def findFailureReason(dbModel: DbModel) = {
+    val smsFailureStatuses = Seq(SMSStatus.Failed, SMSStatus.Undelivered)
+    val emailFailureStatuses = Seq(EmailStatus.Bounced, EmailStatus.Dropped)
+    dbModel.events.collectFirst {
+      case FailedEvent(event) => event.reason
+      case SMSProgressedEvent(event) if smsFailureStatuses.contains(event.status) && event.reason.isDefined =>
+        event.reason.get
```
Comment (cb372) [inline app/models/Comm.scala]: :rage4:

Comment (cb372) [inline app/models/Comm.scala]: How about this?

```scala
dbModel.events.collect {
  case FailedEvent(event) => Some(event.reason)
  case SMSProgressedEvent(event) if smsFailureStatuses.contains(event.status) => event.reason
  case EmailProgressedEvent(event) if emailFailureStatuses.contains(event.status) => event.reason
}.flatten.headOption
```

File: app/models/Comm.scala [hunk 2]
```diff
@@ -166,8 +166,16 @@ object Comm {
     case LinkClickedEvent(event) => Click(event.url, OffsetDateTime.parse(event.metadata.createdAt))
   }
 
-  private def findFailed(dbModel: DbModel) = dbModel.events.collectFirst {
-    case FailedEvent(event) => event
+  private def findFailureReason(dbModel: DbModel) = {
+    val smsFailureStatuses = Seq(SMSStatus.Failed, SMSStatus.Undelivered)
```
Comment (cb372) [inline app/models/Comm.scala]: These vals could be fields, no need to recreate them every time you call the method. Also could be sets.

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 180](https://circleci.com/gh/ovotech/comms-audit-log/180) at 2017-05-02 10:06 BST.

**Pull request merged**